### ISSUE TITLE
Add support for external LanguageTool servers

### ISSUE THREAD

**Issue by valentjn** [valentjn/vscode-ltex#36]
**Is your feature request related to a problem? Please describe.**
More advanced users might want to have their own instances of LanguageTool running on their local machines. For instance, in #34, there was a bug in LT that has been fixed in the development snapshot. Impatient users could run the LT server from the development snapshot for the fix, rather than having to wait for the next LT and LT<sub>E</sub>X release.

**Describe the solution you'd like**
Instead of using the built-in version of LT, connect to an LT server on localhost or wherever.

**Describe alternatives you've considered**
None.

**Additional context**
None.

**Comment by Zplusless** [comment]
`start LT if it not running` will be a good feature. Like Texstudio

**Comment by valentjn** [comment]
@Zplusless The newly implemented solution relies on the user starting the server, as mostly experts will use the server feature anyway. TeXstudio is a different story as it's written in C++ and thus needs to use LT as a server, in contrast to LT<sub>E</sub>X's Java core which can directly interact with LT.

**Comment by valentjn** [comment]
Fix released in 4.9.2.

**Comment by valentjn** [comment]
Reopening as the implementation depended on Java 11, which broke LT<sub>E</sub>X for some users (see #38). It's possible to implement this using a dependency, so this feature does not really on Java 11. Reverted this for now in 4.9.3.

**Comment by ghost** [comment]
I'm currently running a server locally with docker for LanguageTools extension to not rely on external service. I'd like to not run the service multiple times :smile:

**Comment by valentjn** [comment]
I have reimplemented this using Java 8. Still needs to be tested though. It should make it to the 5.0 release.

**Comment by valentjn** [comment]
The new implementation is now in `master` and can be tested via 5.0.0-beta.1. Seems to work for me. Unit tests still need to be implemented.

**Comment by St-Ex** [comment]
I will try it. Thanks.

For information, here how I run a local languageTool server

```console
docker run --rm -p 9090:8010 --name languageTools erikvl87/languagetool
```

Server adress is http://127.0.0.1:9090

**Comment by ghost** [comment]
I will try it. Thanks.

For information, here how I run a local LanguageTool server

```console
docker run --rm -p 9090:8010 --name languageTools erikvl87/languagetool
```

Server address is http://127.0.0.1:9090

**Comment by valentjn** [comment]
Fix released in 5.0.0.
